verification/getCertificateChain_test.go

```diff
@@ -0,0 +1,184 @@
+		failed = true
+	}
+
+	// zlint provides a lot of linter sources. Limit results to just the relevant RFCs.
+	// For a full listing of supported linter sources, see https://github.com/zmap/zlint/blob/master/v3/lint/source.go
+	registry, err := lint.GlobalRegistry().Filter(lint.FilterOptions{
+		IncludeSources: lint.SourceList{
+			lint.RFC3279,
+			lint.RFC5280,
+			lint.RFC5480,
+			lint.RFC5891,
+			lint.RFC8813,
+		}})
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not set up zlint registry: %v", err)
+	}
+
+	for _, cert := range certs {
+		results := zlint.LintCertificateEx(cert, registry)
+
+		for id, result := range results.Results {
+			var level string
+			switch result.Status {
+			case lint.Error:
+				level = "ERROR"
+			case lint.Warn:
+				level = "WARN"
+			default:
+				continue
+			}
+			details := result.Details
+			if details != "" {
+				details = fmt.Sprintf("%s. ", details)
+			}
+			l := registry.ByName(id)
+			// TODO(https://github.com/chipsalliance/caliptra-dpe/issues/74):
+			// Fail the test with Errorf here once we expect it to pass.
+			t.Logf("[%s] %s: %s%s (%s)", level, l.Source, details, l.Description, l.Citation)
+			failed = true
+		}
+
+		if failed {
+			// Dump the cert in PEM and hex for use with various tools
+			t.Logf("[LOG]: Offending certificate (PEM):\n%s", (string)(pem.EncodeToMemory(&pem.Block{
+				Type:  "CERTIFICATE",
+				Bytes: certData,
+			})))
+			t.Logf("[LOG]: Offending certificate (DER):\n%x", certData)
+		}
+	}
+
+	validateCertChain(t, x509Certs, nil)
+	return x509Certs
+}
+
+// Build certificate chain and calls to validateSignature on each chain.
+func validateCertChain(t *testing.T, certChain []*x509.Certificate, leafCert *x509.Certificate) {
+	t.Helper()
+	var certsToProcess []*x509.Certificate
+	if leafCert != nil {
+		t.Log("[LOG]: Validating leaf certificate chain...")
+		certsToProcess = []*x509.Certificate{leafCert}
+	} else {
+		t.Log("[LOG]: Validating intermediate certificates chains...")
+		certsToProcess = certChain
+	}
+
+	// Remove unhandled critical extensions reported by x509 but defined in spec
+	t.Log("[LOG]: Checking for unhandled critical certificate extensions unknown to DPE certificates profile spec...")
+	removeTcgDiceCriticalExtensions(t, certsToProcess)
+
+	// Remove unhandled extended key usages reported by x509 but defined in spec
+	t.Log("[LOG]: Checking for extended key usages unknown to DPE certificates profile spec...")
+	removeTcgDiceExtendedKeyUsages(t, certsToProcess)
+
+	// Build verify options
+	opts := buildVerifyOptions(t, certChain)
+
+	if leafCert != nil {
+		// Certificate chain validation for leaf
+		chains, err := leafCert.Verify(opts)
+		if err != nil {
+			// Certificate chain cannot be built from leaf to root
+			t.Errorf("[ERROR]: Error in certificate chain %s: ", err.Error())
+		}
+
+		// Log certificate chains linked to leaf
+		t.Logf("[LOG]: Chains of DPE leaf certificate.")
+		if len(chains) == 0 {
```

Comment:
It is also invalid if len(chains) > 1. The cert chain should only contain one chain from leaf to root.